### ISSUE TITLE
Article View that was opened via a link from Article Category List: Module loaded twice

### ISSUE THREAD

**Issue by sandewt** [joomla/joomla-cms#18016]
### Steps to reproduce the issue
If you open an **article**, wich is part of an **Article Category List** _and_ in wich a **module** is loaded with **loadposition**, the module _is loaded twice_. This can break (custom) code in Joomla! 3.8.0.

- Load p.e. the module Latest User in an article.
- Use the {loadpostion xxxx} method.
- This article should be part of an Article Category List.
- Open the link to this article via the Article Category List.
- Open the Joomla! Debug Console > Profile information (settings in the Global Configuration).
- See that the module mod_user_latest is loaded twice.

### Expected result

- Module should be loaded once. 

### Actual result

- Module is loaded twice.

### System information (as much as possible)

- Joomla! 3.8.0
- Windows 10
- Xampp

### Additional comments

See image: example J3.8.0

![debug](https://user-images.githubusercontent.com/9271775/30650658-bed65f42-9e23-11e7-973f-58bfd03da12b.png)

Attention to: (2 x 2)
Application: beforeRenderModule mod_users_latest (Users Latest (copy))
Application: afterRenderModule mod_users_latest (Users Latest (copy))
 
In Joomla! 3.7.5 the module is loaded once. (OK)

**Comment by ghost** [comment]
looks similar to #17999 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18016">issues.joomla.org/tracker/joomla-cms/18016</a>.</sub>

**Comment by ggppdk** [comment]
Reason seems to be that 'onContentPrepare' is trigger twice
once for `->text` and once for `->introtext`

https://github.com/joomla/joomla-cms/pull/17175

see also
https://github.com/joomla/joomla-cms/issues/17830

I don't say that triggering plugins twice for same text is bad,
thus is do not say that PR 17175 is bad

- but several plugins are not designed to be triggered twice for same text

e.g. in this issue we don't have bad side effects

and yes issue #17999 looks same as this one but in that case the double loaded module has some bad side-effects

**Comment by ghost** [comment]
@ggppdk so i close this Issue as duplicated Report?

**Comment by ggppdk** [comment]
@franz-wohlkoenig 

issue #17999 shows an important side effect of double loading of modules as a result of the double plugin triggering on same text

but then we have another side-effect
- issue #17830 

i cannot say to close any of them, since people will resubmit them ?

Also the title of this issue is wrong / misleading , and should be fixed !

This issue is about article view that was opened via a link from Article Category List 
So this issue is about article view !

**Comment by ghost** [comment]
@ggppdk please have a Look at updated Title.

**Comment by sandewt** [comment]
Indeed, I have tested using the js code from issue #17999, the title should be about article view.
The debugger shows sometimes other results !?
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18016">issues.joomla.org/tracker/joomla-cms/18016</a>.</sub>

**Comment by sandewt** [comment]
For me is it an issue, because functions will be double loaded, which leads to a fatal error: Cannot redeclare myfunction() / previously declared in...

With the following restriction, it concerns own code!
I solve this problem with the following code: _if (!function_exists('myfunction')) {_ etc.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18016">issues.joomla.org/tracker/joomla-cms/18016</a>.</sub>

**Comment by aardgoose** [comment]
Plugins are called twice where the complete text is displayed and **no** intro text is defined by <read more> markers etc.

 This is certainly an plugin API change, and in my case breaks a site, and is adding extra processing server side for all pages.

#17175 is problematical and changes the plugin API.

**Comment by chiefguru61** [comment]
This has broken my site and caused me three days of problems trying to workout why some custom code loaded in articles via jumi and {loadposition xxx} didn't work and other bits did.

Data saved in the session was being processed twice.

I have resolved the issue by moving my custom code out of the article an into another position on the page.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18016">issues.joomla.org/tracker/joomla-cms/18016</a>.</sub>

**Comment by regularlabs** [comment]
This is a serious issue.
Content plugins are now pretty much all triggered twice.
This causes a whole range of issues. The least bad being that pages load slower, as plugins are doing the same thing twice.
The worst case scenario is that fatal errors happen because of things being triggered/executed twice that shouldn't be.

When I remove the code added through #17175 the issue is gone.
That change should be reverted asap, as it is a B/C change!

**Comment by laoneo** [comment]
If #17175 get reverted then we need to find a solution how to prepare the intro text. @regularlabs what do you suggest?

**Comment by regularlabs** [comment]
I guess this is needed for the custom fields?
Solve it in the custom fields plugin.

**Comment by regularlabs** [comment]
The code in #17175 assumes plugins are not already doing stuff to the intro text.
But the entire `$item` is passed. So plugins already have the power and ability to do stuff separately on the `$item->introtext`, `$item->fulltext` and `$item->text`.
And many plugins are already doing that.

The `onContentPrepare` event doesn't expect a single string and do stuff on there. It handles a complete article object.
#17175 messes with the values of the article object and just runs `onContentPrepare` again.

**Comment by regularlabs** [comment]
Example of what plugins do inside the `onContentPrepare`:
```
public function onContentPrepare($context, &$article, &$params)
{
	if (isset($article->title))
	{
		doStuffWithContent($article->title);
	}

	if (isset($article->description))
	{
		doStuffWithContent($article->description);
	}

	if (isset($article->text))
	{
		doStuffWithContent($article->text);

		// don't also do stuff on introtext/fulltext if text is set
		return;
	}

	if (isset($article->introtext))
	{
		doStuffWithContent($article->introtext);
	}

	if (isset($article->fulltext))
	{
		doStuffWithContent($article->fulltext);
	}
}
```

**Comment by laoneo** [comment]
Ok, got it, will prepare a pr over the weekend.

**Comment by regularlabs** [comment]
Ok, great :)

**Comment by laoneo** [comment]
Please test #18066, it should fix the issue.

**Comment by ghost** [comment]
closed as having Pull Request #18066<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18016">issues.joomla.org/tracker/joomla-cms/18016</a>.</sub>

**Comment by sandewt** [comment]
Thanks all.
